### PR TITLE
Development

### PR DESCRIPTION
**Pull request by SAYUK09**
Please review and provide feedback.

[Live Site](https://podcart.netlify.app/)

### CODE REVIEW

**Comment by soham-parab** [comment]
Great work Sayuri! All functionalities work as expected, UI looks clean and interactive.

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { Router } from "./components/routes/Router";
+// import {
+//   BrowserRouter as Router,
+//   Routes,
+//   Route,
+//   Navigate,
+//   useNavigate,
+//   useParams,
+//   useLocation
+// } from "react-router-dom";
```
Comment (soham-parab) [inline src/App.js]: Unused imports can be discarded instead of commented out.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { Products } from "./pages/products/Products";
+import { Cart } from "./pages/cart/Cart";
+import { Nav } from "./components/nav/Nav";
+import { Wishlist } from "./pages/wishlist/Wishlist";
+import { ProductDetail } from "./pages/product-details/ProductDetail";
+
+export default function App() {
+  const { state, dispatch } = useProductContext();
+  const { cartState, cartDispatch } = useCart();
+  const [route, setRoute] = useState("product");
```
Comment (soham-parab) [inline src/App.js]: This hook can be discarded as it's unused.

File: src/api/server_mock.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          image: faker.random.image(),
+          price: faker.commerce.price(),
+          quantity: 1,
+          isInCart: true,
+          isThereInWishList: false,
+        });
+      });
+    }
+  });
+}
```
Comment (soham-parab) [inline src/api/server_mock.js]: Mock server using mirageJS is unused.

File: src/components/product-detail-card/Product-Detail-Card.css [hunk 4]
```diff
@@ -0,0 +1,107 @@
+.prodBody {
+  display: flex;
+  /* justify-content: space-evenly; */
```
Comment (soham-parab) [inline src/components/product-detail-card/Product-Detail-Card.css]: Comments in CSS files can cause confusion to a collaborator.

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  useEffect(() => {
+    (async function () {
+      try {
+        const response = await axios.get(
+          "https://lazytalkativeinstitutions.sayuk.repl.co/products"
+        );
+
+        console.log(response.data);
```
Comment (soham-parab) [inline src/App.js]: `console.log()` can be removed.

File: src/components/routes/index.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export { Routess } from "./Router";
```
Comment (soham-parab) [inline src/components/routes/index.js]: This file can be removed as its not being used.

File: src/components/sidebar/sidebar.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import { useProductContext } from "../../contexts/Products-Context";
+
+export function Sidebar({}) {
```
Comment (soham-parab) [inline src/components/sidebar/sidebar.js]: The sidebar function doesn't accept any props and hence `{}` can be removed

File: src/components/sidebar/sidebar.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        </label>
+      </div>
+
+      <div className="asideTitle">PRICE RANGE</div>
+      <label>
+        <input
+          className="slider"
+          type="range"
+          min="1200"
+          max="100000"
```
Comment (soham-parab) [inline src/components/sidebar/sidebar.js]: Range for item prices can be evenly distributed across the slider percentage.

File: src/reducers/Cart-Reducer.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+export function cartReducer(cartRedcState, action) {
+  switch (action.type) {
+    case "SET_CART":
+      return { ...cartRedcState, cart: action.payload };
+      break;
```
Comment (appsplash99) [inline src/reducers/Cart-Reducer.js]: when a function *returns* something, the flow of execution exits the function\
and regardless of the whatever is present below/after the *return xyz* statement,\
the code below return never gets executed.

So, no need to write break statement.

File: src/reducers/Products-Reducer.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
+export function productsReducer(RedcState, action) {
+  switch (action.type) {
+    case "PRODUCTS":
+      return { ...RedcState, products: action.payload };
+      break;
```
Comment (appsplash99) [inline src/reducers/Products-Reducer.js]: same as previous, remove *break* after return *statement*

File: src/reducers/Products-Reducer.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    case "PRICE_RANGE":
+      return { ...RedcState, priceRange: action.payload };
+      break;
+
+    case "RESET":
+      return { ...RedcState, sortBy: "none", priceRange: 0 };
+      break;
+
+    case "MICS":
+      console.log(action.isCheck);
```
Comment (soham-parab) [inline src/reducers/Products-Reducer.js]: Unused cases can be removed.

File: src/components/cart-product-card/Cart-Product-Card.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+        </div>
+        <div className="HorizCardDetails">
+          <div className="brandTitle">{prd.name}</div>
+          <p className="prdDescrip">{prd.brand}</p>
+          <div className="horizCardBtnDiv">
+            <button
+              className="HorizQtyBtn"
+              onClick={() => {
+                axiosAddQty(prd, cartDispatch);
+                // cartDispatch({ type: "INCREMENT", payload: prd });
```
Comment (appsplash99) [inline src/components/cart-product-card/Cart-Product-Card.js]: please remove this, if not needed.

File: src/reducers/Cart-Reducer.js (hunk 9 above)
Comment (soham-parab) [inline src/reducers/Cart-Reducer.js]: `SET_CART` can be named better, for example, `LOAD_CART`

File: src/components/cart-product-card/Cart-Product-Card.js [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+          className="horizFooterBtn secBtn"
+          onClick={() => {
+            axiosRemoveFromCart(prd, cartDispatch);
+          }}
+        >
+          Remove
+        </button>
+        <button
+          onClick={() => {
+            // setwishlistItems((prds) => [...prds, prd]);
```
Comment (appsplash99) [inline src/components/cart-product-card/Cart-Product-Card.js]: please remove these comments

File: src/components/product-detail-card/Product-Detail-Card.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+          </p>
+
+          <small>Category : {category}</small>
+        </div>
+        <div className="prodDetailFooter">
+          <button
+            className="prodPrimaryBtn"
+            onClick={() => {
+              axiosAddToCart(prd, cartDispatch);
+              console.log("lala");
```
Comment (appsplash99) [inline src/components/product-detail-card/Product-Detail-Card.js]: unwanted console.log, can be removed

File: src/components/nav/Nav.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+          <img className="navLogo" src="/ecom-nav-favicon.png" />
+        </div>
+        <input className="navSearchBar" placeholder="🔎Search something" />
+        <div className="navIcons">
+          <Link className="routeLink" to="/">
+            <CgMediaPodcast />
+          </Link>
+          &nbsp;
+          <Link className="routeLink" to="/cart">
+            <span className="iconBadge">{cartState.cart.length}</span>
```
Comment (appsplash99) [inline src/components/nav/Nav.js]: Currently, in your UI - navbar's icon:\
if your data is 0 - the badge is still visible;\
what you can try is - when the data is 0 - make the badge to *display: none*

File: src/pages/product-details/ProductDetail.js [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+import { useWishlist } from "../../contexts/Wishlist-Context";
+
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  Navigate,
+  useNavigate,
+  useParams,
+  useLocation
```
Comment (soham-parab) [inline src/pages/product-details/ProductDetail.js]: Unused imports.

File: src/components/cart-product-card/Cart-Product-Card.js [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+            // removeFromCart(prd);
+            axiosAddToWishlist(prd, wishlistItems, setwishlistItems);
+            axiosRemoveFromCart(prd, cartDispatch);
+            // cartDispatch({ type: "REMOVE", payload: prd });
+          }}
+          className="horizFooterBtn"
```
Comment (appsplash99) [inline src/components/cart-product-card/Cart-Product-Card.js:68]: 1. currently in the UI,\
after hovering over *wishlist* button,\
only the background color changes\
but after clicking the button - even though the item gets added to wishlist in backend\
but the heart icon on the wishlist does not gets filled after the wishlist item gets added.

2. on hovering over buttons\
add a *cursor: pointer* class just to keep accessibility in mind.

File: src/api/server_mock.js [hunk 18]
```diff
@@ -0,0 +1,36 @@
+import { createServer, Model, RestSerializer } from "miragejs";
```
Comment (appsplash99) [inline src/api/server_mock.js]: if not needed, kindly remove this unwanted file.

File: src/App.js [hunk 19]
```diff
@@ -1,23 +1,23 @@
...
-  BrowserRouter as Router,
-  Routes,
-  Route,
-  Navigate,
-  useNavigate,
-  useParams,
-  useLocation
-} from "react-router-dom";
+import { Router } from "./components/routes/Router";
+// import {
```
Comment (appsplash99) [inline src/App.js]: please remove unwanted comments

File: src/App.js [hunk 20]
```diff
@@ -41,7 +41,9 @@ export default function App() {
 
   return (
     <div className="App">
-      <Router>
+      <Nav />
+      <Router />
```
Comment (appsplash99) [inline src/App.js:31]: Great Way of Abstracting Router into separate component\
to make *App.js* look clean. :thumbsup:

File: src/components/cart-product-card/Cart-Product-Card.js [hunk 21]
```diff
@@ -0,0 +1,78 @@
...
+
+  return (
+    <div key={prd._id} className="horizCardParent">
+      <div className="horizCardBody">
+        <div className="HorizImgDiv">
+          <img src={prd.image} />
+        </div>
+        <div className="HorizCardDetails">
+          <div className="brandTitle">{prd.name}</div>
+          <p className="prdDescrip">{prd.brand}</p>
```
Comment (appsplash99) [inline src/components/cart-product-card/Cart-Product-Card.js:29]: Smart way to truncate the long description text in the UI 👍

File: src/pages/products/Products.js [hunk 22]
```diff
@@ -0,0 +1,97 @@
...
+                  </div>
+                </Link>
+                <div className="cardDetail">
+                  <h3 className="productTitle">{prd.name}</h3>
+                  <h4 className="productDescrip">{prd.description}</h4>
+                  <div className="secondaryBtnDiv">
+                    <button
+                      onClick={() => {
+                        axiosAddToCart(prd, cartDispatch);
+                        // cartDispatch({ type: "ADD_TO_CART", payload: prd });
```
Comment (soham-parab) [inline src/pages/products/Products.js]: Commented code can be removed to avoid confusion.

**Review by appsplash99 — COMMENTED** [review]
Kindly go through the below reviews.
Above all, don't stop building. 😃

**Review by soham-parab — COMMENTED** [review]
Minor changes suggested, overall brilliant work!

File: src/reducers/Cart-Reducer.js (hunk 9 above)
Comment (SAYUK09) [reply src/reducers/Cart-Reducer.js]: Got It,
Thanks for the information.

File: src/components/product-detail-card/Product-Detail-Card.js (hunk 14 above)
Comment (SAYUK09) [reply src/components/product-detail-card/Product-Detail-Card.js]: Yup, need to do that in all files

File: src/components/nav/Nav.js (hunk 15 above)
Comment (SAYUK09) [reply src/components/nav/Nav.js]: Thanks, Added this feature

File: src/api/server_mock.js (hunk 18 above)
Comment (SAYUK09) [reply src/api/server_mock.js]: Thanks for pointing out

File: src/App.js (hunk 20 above)
Comment (SAYUK09) [reply src/App.js:31]: Thanks,
This one of the suggestions in Suyash's Review tho

File: src/components/cart-product-card/Cart-Product-Card.js (hunk 21 above)
Comment (SAYUK09) [reply src/components/cart-product-card/Cart-Product-Card.js:29]: Thank you

File: src/App.js (hunk 1 above)
Comment (SAYUK09) [reply src/App.js]: Yes, need to do that in all files

File: src/App.js (hunk 2 above)
Comment (SAYUK09) [reply src/App.js]: Thanks for pointing out.

File: src/api/server_mock.js (hunk 3 above)
Comment (SAYUK09) [reply src/api/server_mock.js]: Deleted the file, thank you

File: src/components/product-detail-card/Product-Detail-Card.css (hunk 4 above)
Comment (SAYUK09) [reply src/components/product-detail-card/Product-Detail-Card.css]: Thanks, Removed comments from all CSS files

File: src/components/sidebar/sidebar.js (hunk 7 above)
Comment (SAYUK09) [reply src/components/sidebar/sidebar.js]: Thanks, for pointing that out

File: src/reducers/Products-Reducer.js (hunk 11 above)
Comment (SAYUK09) [reply src/reducers/Products-Reducer.js]: Thanks, this is being taken care of now.

File: src/reducers/Cart-Reducer.js (hunk 9 above)
Comment (SAYUK09) [reply src/reducers/Cart-Reducer.js]: That's seems much better

File: src/components/routes/index.js (hunk 6 above)
Comment (SAYUK09) [reply src/components/routes/index.js]: Thanks for pointing out

**Pull request merged**